src/Init/Data/BitVec/Basic.lean

```diff
@@ -575,6 +575,17 @@ def replicate : (i : Nat) → BitVec w → BitVec (w*i)
       rw [Nat.mul_add, Nat.add_comm, Nat.mul_one]
     hEq ▸ (x ++ replicate n x)
 
+/-! ### flatten -/
+
+/-- Flatten a list of bitvectors into one bitvector. -/
+protected def flatten {n : Nat} (xs : List (BitVec n)) : BitVec (n * xs.length) :=
+  match xs with
+  | [] => 0#0
+  | x :: rest =>
+    have h : n + n * List.length rest = n * List.length (x :: rest) := by
+      simp [List.length_cons, Nat.mul_add, Nat.add_comm]
+    BitVec.cast h (x ++ (BitVec.flatten rest))
```

Comment:
I think this should either be `List.flattenBV` or `flattenList`.

I also think this should be made tail recursive.

Reply:
I've renamed to `flattenList`.

I'm a little skeptical of making it tail-recursive. Of course, it's doable, but comes at a time cost, and I'm not sure that long lists of bitvectors arise in practice.